UPGRADE-4.0.md

```diff
@@ -0,0 +1,66 @@
+# Upgrade from v3.x to v4.x
+
+If you have any questions regarding this upgrade process, please consult the [`examples`](https://github.com/terraform-aws-modules/terraform-aws-vpc/tree/master/examples/) directory:
+
+If you find a bug, please open an issue with supporting configuration to reproduce.
+
+## List of backwards incompatible changes
+
+- The minimum required Terraform version is now 1.0
+- The minimum required AWS provider version is now 4.x (4.35.0 at time of writing)
+- `assign_ipv6_address_on_creation` has been removed; use the respective subnet type equivalent instead (i.e. - `public_subnet_assign_ipv6_address_on_creation`)
+- `enable_classiclink` has been removed; it is no longer supported by AWS https://github.com/hashicorp/terraform/issues/31730
+- `enable_classiclink_dns_support` has been removed; it is no longer supported by AWS https://github.com/hashicorp/terraform/issues/31730
+
+## Additional changes
+
+### Modified
+
+- `map_public_ip_on_launch` now defaults to `false`
+- `enable_dns_hostnames` now defaults to `true`
+- `enable_dns_support` now defaults to `true`
+- `manage_default_security_group` now defaults to `true`
+- `manage_default_route_table` now defaults to `true`
+- `manage_default_network_acl` now defaults to `true`
```

Comment:
When trying to run `terraform destroy` in `examples/simple`, I got this warning:
```
│ Warning: EC2 Default Network ACL (acl-0b1010b9914b276fc) not deleted, removing from state
```

Let's keep the defaults to `false` and let the user become an expert before switching these values.

Reply:
see comment here https://github.com/terraform-aws-modules/terraform-aws-vpc/pull/838#discussion_r1110071817

Reply:
Let's update an example not to have a warning on destroy?

Reply:
added example disabling this here https://github.com/terraform-aws-modules/terraform-aws-vpc/pull/838/commits/28fb8fa017efa2cd2c894c3c96e13b66b953d70b